### ISSUE TITLE
Discussion: Align marketplace extension ids

### ISSUE THREAD

**Issue by spoenemann** [VSCodium/vscodium#494]
## Background

When I contributed the initial version of the VS Code adapter of [Open VSX](https://open-vsx.org) to VSCodium (#404), I had used the internal extension ids of Open VSX in the marketplace API used for installing and updating extensions. At that time, I did not know exactly how the update mechanism works and what implications this choice would have. Now I realized that there are some complications because the extension id returned by the marketplace API is used by VS Code / VSCodium to find updates of already installed extensions:

 * The Open VSX extension ids are not compatible with those from the MS Marketplace, so you lose the auto-update functionality when switching between the two marketplaces. Maybe this is the cause of #483.
 * There are situations where it is necessary to change the marketplace id of an extension to a specific value. This is not easy to accomplish with the current implementation. Examples where we need to make sure a new extension uses the same marketplace id as an older version of it:
    * An extension is deleted for some reason (e.g. broken content) and then published again.
    * An extension author wants to change the namespace / publisher (e.g. https://github.com/eclipse/open-vsx.org/issues/136). 

## Proposal

I would like to change the extension ids returned by open-vsx.org to UUIDs instead of the internal ids. We could try to align these UUIDs with those used by the MS Marketplace, and fall back to generated UUIDs where that fails.

## Consequences

:-1: Auto-update of extensions that were previously installed from Open VSX stops working in VSCodium after the change is deployed. You need to reinstall these extensions to update them once, afterwards everything is fine again. Extensions installed before switching to Open VSX or after this change can be auto-updated without problems.

:+1: We might be able to align extension ids with the MS Marketplace, allowing to switch between the two marketplaces and keep auto-update working.

:+1: Changing extension ids will be much easier, allowing to republish extensions (e.g. with different namespace) and keep auto-update working.

---

I'm posting this topic here instead of [openvsx](https://github.com/eclipse/openvsx) because the change would affect only users of VS Code forks, of which VSCodium is the most prominent.

@stripedpajamas what do you think?

**Comment by stripedpajamas** [comment]
Seems like a good move to me :smile: 

> Auto-update of extensions that were previously installed from Open VSX stops working in VSCodium after the change is deployed. You need to reinstall these extensions to update them once, afterwards everything is fine again. Extensions installed before switching to Open VSX or after this change can be auto-updated without problems.

Wondering how we can mitigate this one "con". Release notes would be a basic one but very people will see it. I might create a pinned issue about it so if anyone realizes their extensions are not updating they will see it.

**Comment by spoenemann** [comment]
Note that we had the same situation when we switched from MS Marketplace to Open VSX. I did not understand the relevance of the extension ids at that time.

When I implemented this change in https://github.com/eclipse/openvsx/pull/180, I included a fall-back so the correct data is returned even when the registry is queried with an "old" numeric id. But I don't know how VS Code handles the conflict between the installed extension id and the id returned by the query.

I'd like to deploy the change next week, maybe on Tuesday.
